pkg/berglas/list.go

```diff
@@ -15,15 +15,52 @@
 package berglas
 
 import (
+	"cloud.google.com/go/storage"
 	"context"
 	"sort"
+	"time"
 
 	"github.com/pkg/errors"
 	"google.golang.org/api/iterator"
 )
 
+// Secret represents a specific secret stored in Google Cloud Storage
+// The attributes on this object should ideally map 1:1 with storage.ObjectAttrs
+type Secret struct {
+	// Name of the secret
+	Name string
+
+	// UpdatedAt indicates when a secret was last updated
+	UpdatedAt time.Time
+
+	// Generation indicates a secret's version
+	Generation int64
+}
+
+// SecretSlice is a list of secrets
+type SecretSlice []Secret
+
+// Len is the number of elements in the collection.
+func (s SecretSlice) Len() int {
```

Comment:
tbh you can use sort.Slice and make some type assertions to end up with fewer methods if you want https://godoc.org/sort#Slice

Reply:
honestly, I did write it like that at first, but I don't like the signature of that goroutine since it needs to read variables outside it's scope.